AP_Baro: support for barometer position compensation

ahmedmsalah99 opened the pull request:
Addressing: #30727

Added baro position offset simulation and variables SIM_BARO_POS_*. 
Added baro position offset variables BARO_POS_* in the barometer implementation.

Added correction for barometer offset in both EKF2 and EKF3.

ahmedmsalah99 commented:
Test:

After launching a plane SITL simulation with a normal mission

wp load ../Tools/autotest/Generic_Missions/CMAC-circuit.txt

param set SIM_BARO_POS_X 60
param set SIM_BARO_POS_Y 60
param set SIM_BARO_POS_Z 10

mode auto
arm throttle

then mid flight set BARO_POS_* to same values as above


results for EKF2 and EKF3 show big fluctuations in estimation when compensation variables are not set correctly which is corrected when the parameters are set:

 <img width="1200" height="563" alt="EKF2_Pos_D" src="https://github.com/user-attachments/assets/7212618b-f0ac-4e30-85e8-65170f725b85" />

<img width="1200" height="563" alt="EKF3_State_Values_Pos_D" src="https://github.com/user-attachments/assets/e5a1e3fc-322e-44ed-8a3d-68099c16cce3" />

peterbarker commented on libraries/AP_Baro/AP_Baro.cpp:
```diff
@@ -229,6 +229,13 @@ const AP_Param::GroupInfo AP_Baro::var_info[] = {
 #endif
 #endif  // HAL_BARO_WIND_COMP_ENABLED
 
+    // @Param: BARO_POS
```
```suggestion
#if AP_BARO_POSITION_COMPENSATION_ENABLED
    // @Param: BARO_POS
```

... this should be a zero-compiler-change output on 1MB boards.  And definitely off on AP_Periph.

peterbarker commented:
@Williangalvani will be interested in this

peterbarker commented on libraries/AP_Baro/AP_Baro.h:
```diff
@@ -251,7 +254,7 @@ class AP_Baro
 
     uint8_t msp_instance_mask;
-
+    AP_Vector3f pos_offset;
```
```suggestion
#if AP_BARO_POSITION_COMPENSATION_ENABLED
    AP_Vector3f pos_offset;
#endif  // AP_BARO_POSITION_COMPENSATION_ENABLED
```

... and around the new methods and whatever else is being introduced.  THis should be a no-compiler-output change if the feature is not enabled.  You only do this in the main code, not in the SITL stuff.

peterbarker commented on libraries/AP_NavEKF2/AP_NavEKF2_core.h:
```diff
@@ -449,6 +449,7 @@ class NavEKF2_core : public NavEKF_core_common
 
     struct baro_elements : EKF_obs_element_t {
         ftype       hgt;
+        bool corrected;
```
I don't think we need to update NavEKF2 for this.

We don't touch EKF2 much, just to keep it as a very stable thing.

peterbarker commented on libraries/SITL/SIM_Aircraft.cpp:
```diff
@@ -698,6 +698,7 @@ void Aircraft::update_model(const struct sitl_input &input)
     }
 }
 
+
```
Stray whitespace change

peterbarker commented on libraries/AP_NavEKF3/AP_NavEKF3_PosVelFusion.cpp:
```diff
@@ -429,6 +429,29 @@ void NavEKF3_core::CorrectGPSForAntennaOffset(gps_elements &gps_data) const
+/*
+  correct Baro data for position offset relative to the IMU
+ */
+void NavEKF3_core::CorrectBaroOffset(baro_elements &baro_data) const
```
```suggestion
void NavEKF3_core::CorrectBaroPositionOffset(baro_elements &baro_data) const
```

"offset" can mean many things

peterbarker commented on libraries/SITL/SITL.cpp:
```diff
@@ -248,6 +248,12 @@ const AP_Param::GroupInfo SIM::var_info[] = {
+    // @Description: XYZ position of the barometer relative to the body frame origin
+    // @Units: m
+    // @Vector3Parameter: 1
+    AP_GROUPINFO("BARO_POS",      57, SIM,  baro_pos_offset, 0),
```
Is it just me or are we missing some code to use this?

ahmedmsalah99 replied:
The simulation code exists and I believe it's working well. 
The changes are in libraries/AP_Baro/AP_Baro_SITL.cpp

peterbarker reviewed (changes requested):
It would be nice to have a simple autotest which tests the new parameters work.

Should be able to set the sim parameter and have the vehicle's reported altitude change, then change the normal parameter and have it return to its normal value.

ahmedmsalah99 committed:
AP_Baro: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

fixing logs

Tools: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

fixing logs

ahmedmsalah99 committed:
AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

fixing logs

Tools: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

fixing logs

ahmedmsalah99 committed:
AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

fixing logs

Tools: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

fixing logs

ahmedmsalah99 committed:
Tools: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

fixing logs

Tools: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

fixing logs

ahmedmsalah99 committed:
SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

fixing logs

Tools: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_DAL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF2: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

AP_NavEKF3: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

SITL: add baro position offset simulation and variables SIM_BARO_POS_*. Add baro position offset variables BARO_POS_* in the barometer implementation.
Added correction for barometer offset in both EKF2 and EKF3.

removing stray whitespaces. Changing logging type to J instead of I. Adding If statement to the EKF3 function and changing its name

Adding infrequent logging and adding #if staments.

fixing some mistakes

adding some ifs and fixing some unexpected changes

removing stray space and fixing EKF2 stuff

removing stray space and fixing EKF2 stuff

adding logic correction

final tweaks

fixing issue

fixing logs

Williangalvani commented:
I agree with @peterbarker .
We could have an autotest that sets SIM_BARO_X and then pitches down, then validate that the baro readings show a higher altitude. On a second part of the test, we can add BARO_OFS_X and validate that it corrects for that offset
